src/course/views.py

```diff
@@ -134,7 +132,67 @@ def post(self):
                 current_user.courses.append(course)
                 models.db.session.commit()
                 author = models.User.query.filter_by(id=course.teacher_id).first()
-                print course.id
                 return "Redirect to:%s" % (url_for("view_course", courseID=course.id+1000))
         else:
             return "Registration Code Incorrect"
+
+class AddTAView(MethodView):
+
+    def get(self):
+        return
+
+    def post(self):
+        data = flask.request.get_json()
+        email = data.get('email')
+        courseId = data.get('courseID')
+        course = models.Course.query.filter_by(id=courseId).first()
+        if email:
+            user = models.User.query.filter(models.User.email.contains(email)).first()
+            if user and user.permissions:
+                    if user.permissions < 20:
+                        user.permissions = 20
+                    secondaryTeachers = course.secondaryTeachers
+                    userIdAndComma = str(user.id) + ","
+                    if secondaryTeachers.find("," + userIdAndComma) >= 0:
+                        return "failure"
+                    secondaryTeachers = secondaryTeachers + userIdAndComma
+                    course.secondaryTeachers = secondaryTeachers
+                    models.db.session.commit()
+                    return email
+            else: 
+                return "failure"
+        else:
+            return "failure"
+
+class RemoveTAView(MethodView):
+
+    def get(self):
+        return
+
+    def post(self):
+        data = flask.request.get_json()
+        email = data.get('email')
+        courseId = data.get('courseID')
+        course = models.Course.query.filter_by(id=courseId).first()
+        if email:
+            user = models.User.query.filter(models.User.email.contains(email)).first()
+            if user and user.permissions:
+                newUser = models.User.query.filter_by(email=email).first()
+                tas = models.Course.query.filter(models.Course.secondaryTeachers.contains(","+str(newUser.id)+",")).all()
```

Comment:
Instead you should just filter by contains(user.id) then if necessary you can remove the improper duplications.

```python
courses_where_maybe_ta = models.Course.filter(models.Course.secondaryTeachers.contains(str(user.id))
courses_where_ta = []
for c in courses_where_maybe_ta:
     if user.id in c.secondaryTeachers.split(', '):
         courses_where_ta.append(c)
```